tests/Doctrine/Tests/ORM/Functional/OneToManyMatchingCriteriaTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\ORM\Mapping\FetchMode;
+use Doctrine\Tests\Models\ECommerce\ECommerceCategory;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Tests sorting mechanism of Selectable::matching()
+ */
+class OneToManyMatchingCriteriaTest extends OrmFunctionalTestCase
```

Comment:
Please use the github issue id as class name, to help us identify things better (GH*)

Reply:
Done, it's GH7767Test now. Moved it to "Functional/Ticket" because that's where other tests seem to live.